### PR TITLE
Logging platform level docs

### PR DESCRIPTION
**Pull request by fhennig**
I want to link to the demo, so that PR needs to be merged first https://github.com/stackabletech/stackablectl/pull/187

### CODE REVIEW

File: modules/concepts/pages/logging.adoc [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+[#configuration]
+== Identical configuration in each product
+
+Logging configuration usually requires configuration what to log, where to log it and in which format. Stackable provides a unified format and also configures where to log - either to the container console or the aggregator. The only thing left to configure is the _what_. As part of the logging configuration, individual modules can be configured with different log levels:
+
+```yaml
+logging:
+  containers:          // <1>
+    mainContainer:
```
Comment (siegfriedweber) [inline modules/concepts/pages/logging.adoc:34]: This must be a valid container name matching the regular expression `[a-z0-9]([-a-z0-9]*[a-z0-9])?`.

```suggestion
    main-container:
```

File: modules/concepts/pages/logging.adoc [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      file:
+        level: INFO
+      loggers:         // <3>
+        ROOT:
+          level: INFO
+        my.module:
+          level: DEBUG
+        some.chatty.module:
+          level: NONE
+    sidecarContainer:
```
Comment (siegfriedweber) [inline modules/concepts/pages/logging.adoc:46]: This must be a valid container name:

```suggestion
    sidecar-container:
```

File: modules/concepts/pages/logging.adoc [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      ...
+```
+
+<1> Logging is defined for each container
+<2> Console and file (aggregator) have different log levels
```
Comment (siegfriedweber) [inline modules/concepts/pages/logging.adoc:51]:
```suggestion
<2> Console and file appenders can have different log level thresholds.
```

File: modules/concepts/pages/logging.adoc [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+<3> Setting log levels for individual modules is also possible
+
+**Log levels per module** - In the `loggers` section the log levels for each module are configured. Note the `ROOT` module. This is a special module and refers to the default log level for the container. The log levels are the commonly used ones (TRACE, DEBUG, INFO, WARN, FATAL) as well as the NONE level to turn of logging.
```
Comment (siegfriedweber) [inline modules/concepts/pages/logging.adoc:54]:
```suggestion
**Log levels per module** - In the `loggers` section the log levels for each module are configured. Note the `ROOT` module. This is a special module and refers to the default log level for the container. The log levels are the commonly used ones (TRACE, DEBUG, INFO, WARN, ERROR, FATAL) as well as the NONE level to turn of logging.
```

File: modules/concepts/pages/logging.adoc [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+image::logging_architecture.drawio.svg[]
+
+You configure your logging in the resource describing the cluster, seen in the top left in the diagram (see the <<configuration, configuration>> section below). The operator reads this resource and creates the appropriate log configuration files as ConfigMaps, which are then mounted into the containers. The ZooKeeper Pod has three containers: The `prepare` sidecar container, the `zookeeper` container and the `vector` https://vector.dev/docs/setup/deployment/roles/#sidecar[sidecar container]. All logs get written into a shared mounted directory, from which the Vector agent reads them and sends them to the vector https://vector.dev/docs/setup/deployment/roles/#aggregator[aggregator].
+
+The aggregator is configured to use the OpenSearch sink. It sends all logs to the sink. If the sink is unavailable, it will buffer the log messages. It is also the single point where the sink is configured, so the sink is decoupled from the individual product configurations and only needs to be configured in this single location for the whole platform.
+
+[#configuration]
+== Identical configuration in each product
```
Comment (siegfriedweber) [inline modules/concepts/pages/logging.adoc:27]: From this documentation, it is not clear to me that the logging configuration is located in the role or rolegroup config.

File: modules/concepts/pages/logging.adoc [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+== Architecture
+
+Below you can see the overall architecture using ZooKeeper as an example. Stackable uses https://vector.dev/[Vector] for log aggregation and https://opensearch.org/[OpenSearch] as the sink to persist logs in.
+
+image::logging_architecture.drawio.svg[]
+
+You configure your logging in the resource describing the cluster, seen in the top left in the diagram (see the <<configuration, configuration>> section below). The operator reads this resource and creates the appropriate log configuration files as ConfigMaps, which are then mounted into the containers. The ZooKeeper Pod has three containers: The `prepare` sidecar container, the `zookeeper` container and the `vector` https://vector.dev/docs/setup/deployment/roles/#sidecar[sidecar container]. All logs get written into a shared mounted directory, from which the Vector agent reads them and sends them to the vector https://vector.dev/docs/setup/deployment/roles/#aggregator[aggregator].
```
[the remark this answers is not in the document]
Comment (fhennig) [reply modules/concepts/pages/logging.adoc]: I've changed it

File: modules/concepts/pages/logging.adoc (hunk 5 above)
Comment (fhennig) [reply modules/concepts/pages/logging.adoc:27]: I've added a paragraph about this, linking to the role and role group docs

File: modules/concepts/pages/logging.adoc [hunk 7]
```diff
@@ -0,0 +1,86 @@
+= Logging
+
+// Abstract
+Logging is important for observability of the platform. Stackable provides human-readable plaintext logs for each running container, as well as aggregated and persisted logs with identical structure across the whole platform. Log levels can be set for individual modules and configuration is identical across all products, but custom logging configuration files can be supplied as well.
+
+image::logging_overview.drawio.svg[]
```
[the remark this answers is not in the document]
Comment (fhennig) [reply modules/concepts/pages/logging.adoc:6]: I agree, doesn't look so nice. Fixed: https://github.com/stackabletech/documentation/pull/326/commits/9bfa44f5e88dfecd76e4609a16f84b0f129518b0

File: modules/concepts/pages/logging.adoc [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+
+[#configuration]
+== Identical configuration in each product
+
+Logging configuration usually requires configuration what to log, where to log it and in which format. Usually you want to persist logs but also view them on the fly in a console, and possibly both of these in different formats. Stackable provides configuration for this out of the box. The only thing left to configure is the _what_. As part of the logging configuration, individual modules can be configured with different log levels:
+
+```yaml
+logging:
+  vectorAggregatorConfigMapName: vector-aggregator-discovery
+  containers:          // <1>
```
[the remark this answers is not in the document]
Comment (fhennig) [reply modules/concepts/pages/logging.adoc]: fixed: 64654df7

**Review by siegfriedweber — APPROVED** [review]
LGTM

**Comment by siegfriedweber** [comment]
@lfrancke, @fhennig fixed this in #332.

**Pull request merged**